src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -146,7 +148,13 @@ internal sealed class SolutionRestoreJob : ISolutionRestoreJob
                 throw new ArgumentNullException(nameof(logger));
             }
 
+            if (vulnerabilitiesFoundService == null)
+            {
+                throw new ArgumentNullException(nameof(vulnerabilitiesFoundService));
```

Comment:
Is it really a critical failure to not find this service? Maybe just log telemetry and make sure to handle the `null`?

Reply:
My point is that this basically disable VS if this MEF import fails. To me, when something is unable to show an infobar, I don't think VS should essentially become inoperable. 
I typically see VS show popups that say "The ______ package failed to load correctly, do you want to continue seeing this message?" If a customer says "no", then restore will now fail because we're throwing `ArgumentNullException` regardless.

Reply:
If MEF imports fail, it'll become unusable anyways. The resolution will make SolutionRestoreWOrker fail, which means restore won't be triggered. 

This is a service we declare, so the expectation should be that it is there.